### PR TITLE
multithreaded benchmarks

### PR DESCRIPTION
**Pull request by jpfairbanks**
This is a better version of #516. Uses PkgBenchmark, adds benchmarks to travis, in a way that won't have benchmarks break the build -- for now.

The benchmarkpkg function asks for user input at the end, how do I get it to say no?

### CODE REVIEW

**Comment by sbromberger** [comment]
Looks like something's timing out looking for input: https://travis-ci.org/JuliaGraphs/LightGraphs.jl/builds/219864431#L1381

File: benchmark/parallel/egonets.jl [hunk 1]
```diff
@@ -0,0 +1,63 @@
+using LightGraphs
```
Comment (sbromberger) [inline benchmark/parallel/egonets.jl]: So, my main concern here is that we're not being forthcoming in what we're actually benchmarking. That is, the other benchmarks, if they fail, will tell us explicitly, "this function has a regression". I'm not sure how a failure here can be interpreted.

Comment (jpfairbanks) [inline benchmark/parallel/egonets.jl]: My intension with this benchmark is to get us to see the parallel performance and track it over time. Not really a hard performance regression detector. 

Of course, if the parallel functions were slower than the serial performance, then I would call that a regression.

**Comment by jpfairbanks** [comment]
The solution is to pass promptoverwrite=false to benchmarkpkg. I think.

**Comment by jpfairbanks** [comment]
https://github.com/JuliaCI/PkgBenchmark.jl/blob/master/src/runbenchmark.jl#L62

**Comment by sbromberger** [comment]
Still getting stuck here: https://travis-ci.org/JuliaGraphs/LightGraphs.jl/builds/220687151#L1397

**Comment by jpfairbanks** [comment]
I think this just needs promptoverwrite set to false https://github.com/JuliaCI/PkgBenchmark.jl#running-a-benchmark-suite

**Comment by sbromberger** [comment]
Can you fix it? I'd really love to merge this.

**Comment by sbromberger** [comment]
If you're good with this, @jpfairbanks, please go ahead and merge. Thank you.

**Pull request merged**